esrally/mechanic/launcher.py

```diff
@@ -208,26 +208,33 @@ def stop(self, nodes, metrics_store):
         else:
             self.logger.info("Shutting down [%d] nodes on this host.", len(nodes))
         for node in nodes:
-            proc = psutil.Process(pid=node.pid)
             node_name = node.node_name
             telemetry.add_metadata_for_node(metrics_store, node_name, node.host_name)
+            try:
+                es = psutil.Process(pid=node.pid)
+                node.telemetry.detach_from_node(node, running=True)
+            except psutil.NoSuchProcess:
+                self.logger.warning("No process found with PID [%s] for node [%s].", node.pid, node_name)
+                es = None
 
-            node.telemetry.detach_from_node(node, running=True)
             if not self.keep_running:
-                stop_watch = self._clock.stop_watch()
-                stop_watch.start()
-                try:
-                    os.kill(proc.pid, signal.SIGTERM)
-                    proc.wait(10.0)
-                except ProcessLookupError:
-                    self.logger.warning("No process found with PID [%s] for node [%s]", proc.pid, node_name)
-                except psutil.TimeoutExpired:
-                    self.logger.info("kill -KILL node [%s]", node_name)
+                if es:
+                    stop_watch = self._clock.stop_watch()
+                    stop_watch.start()
                     try:
-                        # kill -9
-                        proc.kill()
+                        os.kill(es.pid, signal.SIGTERM)
```

Comment:
Doesn't `es.terminate()` kill via SIGTERM? ref: https://psutil.readthedocs.io/en/latest/#psutil.Process.terminate

Reply:
I was only moving this code but have not touched it otherwise. However, I agree this change makes sense. I'll push this in a separate commit.

Reply:
I've pushed https://github.com/elastic/rally/pull/860/commits/32100f528aac4dc20a88f53a4ca568053d8e4555